### ISSUE TITLE
Allow setting primary key for importing in the user info for the entity itself 

### ISSUE THREAD

**Issue by tonyarnold** [magicalpanda/MagicalRecord#752]
As per the suggesting by @andyfitter in #710, this sounds pretty sane to me. I’m not sure if it’s possible, but it’s worth investigating.

**Comment by depl0y** [comment]
Is this already in a release? I got a simple JSON set which defines relationships between objects just because they are included in that object. I do not have a specific ID for the relation ship.

When I import this (as NSDictionary instead of JSON of course), the child objects are duplicated at each import. I did set the `relatedByAttribute` to `id`  on the parent and child entity, but I am not sure this is the way to stop this from happening. The parent objects aren't duplicated but the child objects are.

I do not import the child objects specifically, but I just import the parent object, with a `mappedKeyName` on the Core Data relationship.

Here is an example of my JSON:
```{
  "child_objects" : [
    {
      "id" : 1,
      "name" : "child object 1"
    },
    {
      "id" : 2,
      "name" : "child object 2"
    }
  ],
  "id" : 1,
  "name" : "parent object 1",
  "some_parent_property" : "indeed"
}```

**Comment by andyfitter** [comment]
Tony mentioned this to me a while ago and it seemed to fix my problem. Not the most intuitive option but it fixed it. Maybe it will help? 

"I thought the same thing, but it doesn't work that way right now. I think adding a primaryKey key to the user info of an entity should do the job nicely. I'll have a look at it in the morning if you don't get there first"

**Comment by depl0y** [comment]
Yup, I noticed he said that, but I can't find this option included in the documentation yet. 

So I guess I should use "relatedByAttribute" then.
